builder/processor-tools/src/main/java/io/helidon/builder/processor/tools/BodyContext.java

```diff
@@ -96,16 +101,29 @@ public class BodyContext {
         this.listType = toListImplType(builderTriggerAnnotation, typeInfo);
         this.mapType = toMapImplType(builderTriggerAnnotation, typeInfo);
         this.setType = toSetImplType(builderTriggerAnnotation, typeInfo);
-        gatherAllAttributeNames(this, typeInfo);
+        try {
+            gatherAllAttributeNames(this, typeInfo);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed while processing: " + typeInfo.typeName(), e);
+        }
         assert (allTypeInfos.size() == allAttributeNames.size());
         this.hasParent = Objects.nonNull(parentTypeName.get());
+        this.hasAnyBuilderClashingMethodNames = determineIfHasAnyClashingMethodNames();
+        this.isExtendingAnAbstractClass = typeInfo.typeKind().equals("CLASS");
         this.ctorBuilderAcceptTypeName = (hasParent)
                 ? typeInfo.typeName()
                 : (Objects.nonNull(parentAnnotationType.get()) && typeInfo.elementInfo().isEmpty()
                                 ? typeInfo.superTypeInfo().orElseThrow().typeName() : typeInfo.typeName());
         this.genericBuilderClassDecl = "Builder";
         this.genericBuilderAliasDecl = ("B".equals(typeInfo.typeName().className())) ? "BU" : "B";
         this.genericBuilderAcceptAliasDecl = ("T".equals(typeInfo.typeName().className())) ? "TY" : "T";
+        String interceptorType = searchForBuilderAnnotation("interceptor", builderTriggerAnnotation, typeInfo);
+        this.interceptorTypeName = (Void.class.getName().equals(interceptorType) || Objects.isNull(interceptorType))
+                ? null : DefaultTypeName.createFromTypeName(interceptorType);
+        String interceptorCreateMethod =
+                searchForBuilderAnnotation("interceptorCreateMethod", builderTriggerAnnotation, typeInfo);
+        this.interceptorCreateMethod = ("".equals(interceptorCreateMethod) || Objects.isNull(interceptorCreateMethod))
```

Comment:
please do not use `Objects.isNull` in normal if statements, just do `== null`. This is confusing

Reply:
Also I would start with the null check, so you can then just call `interceptorCreateMethod.isEmpty()`. In general these two cases should not be checked - either we support nulls, and then it should be null. Or we use empty string instead of null, and then it never should be null

Reply:
I changed the order in the next commit; the result of that (private) method can be null if the annotation does not exist, and empty if the annotation exists and is not populated.  But for the initializer here I set it to null as a general (internal) convention.